src/client/info.js

```
@@ -32,10 +32,20 @@ export default (class Info {
                 .get(endpoint, {
                     headers: { 'user-agent': getUserAgent() },
                     followRedirect: false,
+                    retry: {
+                        limit: 0, // no retries
+                    },
+                    timeout: {
+                        connect: 100, // milliseconds
+                    },
                 })
                 .json();
         } catch (err) {
-            return constructError(err);
+            // TODO: was this explicitly returning the error for some reason?
+            // Callers do NOT handle error, so my only assumption is that this
+            // should be exit here
+            const { message } = constructError(err);
+            return printError(`Failed to retrieve cluster information from ${this.cortexUrl}: ${message}`, {}, true);
```

I would like some input on this..

The Info API call was always failing silently. This wasn't really noticeable because either the subsequent compatibilityCheck API call or the Action would also normally fail afterwards in those case.

The differences in user experience are slight and mostly affect the case where the cluster is not reachable. Downside of raising an error here is that `--help` commands will print an error instead of normally exiting, but the upside is that non-help commands will fail faster (no need to wait for the next API failure)

---

So would --help when first installing the CLI (no .cortex file being set) cause an error to print? Seems like the error itself should bubble up.

---

Yes, with these changes using `--help` flag without a profile (`~/.cortex`) does cause an error to print. This wasn't the case before, because API calls weren't made when running help commands. This is one case where the behavior changes.

Yeah, I agree the error shouldn't be always be silent. I might be able to make the error bubble up and force the caller to deal with it, instead of having an hard-exit here. That might help preserve the existing behavior.

---

Looked into this more, and I definitely need to fix this. The current implementation doesn't really let a new user create a profile